scripts/import-skins.ts

```diff
@@ -0,0 +1,308 @@
+ * @returns 6-character HEX color string, or undefined if translation is not possible
+ */
+function translateCssColor(color: string, baseColor: string): string | undefined {
+  // Color format is `fff`
+  if (/^[a-f0-9]{3}$/i.test(color)) {
+    return color
+      .split('')
+      .map(c => c.repeat(2))
+      .join('');
+  }
+
+  // Color format is `ffffff`
+  if (/^[a-f0-9]{6}$/i.test(color)) {
+    return color;
+  }
+
+  // Color format is `white`
+  if (COLOR_MAP[color]) {
+    return COLOR_MAP[color];
+  }
+
+  // Return base color, as transparency is not supported
+  if (color == 'transparent' || color === 'none') {
+    return baseColor;
+  }
+
+  // For all remaining formats, transform to RGBA first, then remove transparency
+  let rgba: number[] = [];
+
+  try {
+    if (/^(rgb|hsl)a?/.test(color)) {
+      // Color format is `rgb()`, `rgba()`, `hsl()` or `hsla()`
+      rgba = extractColorFunctionValues(color);
+    } else if (/[a-f0-9]{8}/.test(color))
+      // Color format is `ffffff22`
+      rgba = hexToRgba(color);
+
+    if (rgba.length !== 4) {
+      throw new Error(`unable to translate color to RGBA: ${color}`);
+    }
+
+    return opacifyColor(rgba, hexToRgba(baseColor || 'ffffff'));
+  } catch (e) {
+    console.log(e);
+  }
+}
+
+async function loadAndConvertSkin(skin: string) {
+  if (skins[skin]) return;
+
+  try {
+    // kebabcase only strings with uppercase, and skip name with number like `tw33t`
+    const skinName = /[A-Z]/.test(skin) ? kebabCase(skin) : skin;
+    const response = await fetch(`${SKINS_ROOT_PATH}${skinName}.scss`);
+    const body = await response.text();
+    const settings = { theme: 'light' };
+
+    if (response.status !== 200) {
+      return;
+    }
+
+    SKIN_COLORS.forEach(key => {
+      const matches = body.match(new RegExp(`--${key}-(color|bg):(?<color>.*);`, 'm'));
+
+      if (!matches) {
+        return;
+      }
+
+      const color = translateCssColor(
+        matches.groups.color.replace('#', '').trim().toLowerCase(),
+        settings['bg_color']
+      );
+      if (!color) return;
+
+      settings[`${key}_color`] = color;
+    });
+
+    if (settings['text_color']) {
+      settings['content_color'] = settings['text_color'];
+      const textRgba = hexToRgba(settings['text_color']);
+      textRgba[3] = 0.85;
+
+      settings['text_color'] = opacifyColor(textRgba, hexToRgba(settings['bg_color'] || 'ffffff'));
+    }
+
+    if (settings['bg_color']) {
+      const [r, g, b] = hexToRgba(settings['bg_color']);
+      settings['theme'] = getBrightness(r, g, b) > 0.5 ? 'light' : 'dark';
+    }
+
+    skins[skin] = settings;
+  } catch (e) {
+    console.log(e);
+  }
+}
+
+async function main() {
+  const startTime = new Date().getTime();
+
+  const spacesWithCustomDomain = await db.queryAsync(`
+    SELECT
+      id,
+      JSON_UNQUOTE(settings->'$.skin') as skin
+    FROM spaces
+    WHERE
+      settings->'$.skin' IS NOT NULL
+      AND domain IS NOT NULL
+  `);
```

Comment:
Doesn't need turbo at this stage?

Reply:
`AND domain IS NOT NULL` is the filter. Skin is enabled on all whitelabel